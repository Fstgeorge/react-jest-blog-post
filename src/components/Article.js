import React, { PropTypes } from 'react';
import { bindAll } from 'lodash';
import ArticleTitle from './ArticleTitle';
import ArticleBody from './ArticleBody';

class Article extends React.Component {

  constructor(props) {
    super(props);
    this.state = {
      isMouseOver: false,
    }
    bindAll(this, ['_onMouseEnter', '_onMouseLeave']);
  }

  _onMouseEnter() {
    this.setState({isMouseOver: true});
  }

  _onMouseLeave() {
    this.setState({isMouseOver: false});
  }

  render () {
    const { article } = this.props;
    return (
      <div
        style={{display: 'inline-block'}} className={this.state.isMouseOver ? 'has-mouse-over' : ''}
        onMouseEnter={this._onMouseEnter} onMouseLeave={this._onMouseLeave}
      >
        <ArticleTitle title={article.title} timestamp={article.timestamp} url={article.url}/>
        <ArticleBody body={article.body} />
      </div>
    );
  }
}

Article.propTypes = {
  article: PropTypes.object.isRequired,
}

export default Article;
