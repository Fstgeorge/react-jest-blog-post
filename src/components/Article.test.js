import renderer from 'react-test-renderer';
import React from 'react';
import Article from './Article';

jest.mock('./ArticleTitle.js');
jest.mock('./ArticleBody.js');

const article = {
  timestamp: '2016-11-12',
  body: 'article body',
  url: 'http://example.com',
  title: 'An Article',
}

test('Article renders correctly', () => {
  const component = renderer.create(
    <Article article={article} />
  );

  let tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback
  tree.props.onMouseEnter();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

  // manually trigger the callback1
  tree.props.onMouseLeave();
  // re-rendering
  tree = component.toJSON();
  expect(tree).toMatchSnapshot();

});
