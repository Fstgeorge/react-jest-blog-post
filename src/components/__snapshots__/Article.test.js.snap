exports[`test renders correctly 1`] = `
<div
  className=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "display": "inline-block",
    }
  }>
  <span>
    ArticleTitle 
    {\"title\":\"An Article\",\"timestamp\":\"2016-11-12\",\"url\":\"http://example.com\"}
  </span>
  <span>
    ArticleBody 
    {\"body\":\"article body\"}
  </span>
</div>
`;

exports[`test renders correctly 2`] = `
<div
  className="has-mouse-over"
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "display": "inline-block",
    }
  }>
  <span>
    ArticleTitle 
    {\"title\":\"An Article\",\"timestamp\":\"2016-11-12\",\"url\":\"http://example.com\"}
  </span>
  <span>
    ArticleBody 
    {\"body\":\"article body\"}
  </span>
</div>
`;

exports[`test renders correctly 3`] = `
<div
  className=""
  onMouseEnter={[Function]}
  onMouseLeave={[Function]}
  style={
    Object {
      "display": "inline-block",
    }
  }>
  <span>
    ArticleTitle 
    {\"title\":\"An Article\",\"timestamp\":\"2016-11-12\",\"url\":\"http://example.com\"}
  </span>
  <span>
    ArticleBody 
    {\"body\":\"article body\"}
  </span>
</div>
`;
